pilot/pkg/networking/core/v1alpha3/listener.go

```diff
@@ -108,6 +111,10 @@ var (
 			"upstream_transport_failure_reason": {Kind: &google_protobuf.Value_StringValue{StringValue: "%UPSTREAM_TRANSPORT_FAILURE_REASON%"}},
 		},
 	}
+
+	// ProxyInboundListenPort is the dedicated inbound traffic capture port for sidecar proxy.
+	// See also `MeshConfig.ProxyListenPort`.
+	ProxyInboundListenPort uint32 = 15006
```

Comment:
Please dont hard code this. We are not hardcoding the proxy listen port. so why is this alone being hardcoded ?

Reply:
My concern is with adding it to values.yaml and exposing it to the user (and not having any test for the case user decides to change it). 
I'm ok with any option that doesn't add more user-visible settings to the already too long list without a clear case that 
it's a common use case to change it - and tests that the change works. 

How about:
- an env variable - would work great on VMs
- a mesh config option ( like the proxy listen port ) - with default value set in the same way ( i.e. hardcoded in the code). 

I would prefer the env because it is not yet clear the final solution will even use this separate port - if Louis doc is right,
as soon as we have FCDS we can move back to single port. Adding it to the API surface has a cost.

Reply:
@rshriram Do you agree if I use env var?

@costinm I am wondering if single port is better even with FCDS. Yes, filterchain support detect if the traffic is from local but it is much more expensive than iptables AFAIK. We can discuss later